### DISCUSSION TITLE
Pore network without throats

### DISCUSSION THREAD

**Discussion by L278** [PMEAL/OpenPNM#1925] [Q&A]
Hi, slightly new to OpenPNM, is it possible to create a network of pores without throats? Also, how would I go about doing that if possible?

**Comment by jgostick** [comment]
Hi @L278 
Technically it's possible but it would not be very useful.  When you generate a network you have 'pore.coords' which are the x,y,z coordinates of each pore, and 'throat.conns' which is a Nx2 array of pore-to-pore connections (i.e. [1, 4] indicates pore 1 is connected to pore 4).  It is entirely possible that pore X has no neighbors, so it would not appear in the throat.conns list (i.e. no entries containing X).  In openpnm we have a check that you can run to ensure this is NOT the case because it breaks the linear solvers.  So, I guess it depends on what your planning to do with your network?  If you just want points in space then you can delete/ignore 'throat.conns', but I don't know what use that would be.

**Reply by L278** [reply]
Hi @jgostick, thanks for your reply. Initially, I was thinking about whether it was possible to make the pores physically connect in space without the need for the throats because I'm trying to model a clayey material, but when you explained it like that, keeping the throats makes more sense and would probably be more useful.

**Reply by jgostick** [reply]
The approach would just be to make the pores large and close to each other so that the throats are short.  You have to be careful though that your 'pore scale models' are able to handle overlap.  For instance, if you have overlapping pores you can end up with negative throat lengths, which cause inaccuracies in the transport modeling since the total length get miscalculated.

**Reply by ma-sadeghi** [reply]
@L278 In case you're trying to do "continuum" modeling using OpenPNM, we have an open pull request (#1825), but it's not quite done yet. Like @jgostick mentioned, it's basically using a cubic network to represent a finite volume grid (i.e., control volume = pore plus zero size throats)

**Reply by L278** [reply]
Hi @ma-sadeghi, thanks for sharing the continuum model, how would I know if it's complete - I wasn't quite sure what the comments, commits, etc meant.

**Reply by ma-sadeghi** [reply]
Well, you need to occasionally check the PR to see if it's been merged. Sorry, we're trying our best to finish it up, but we're a little busy at the moment and so it might take longer.
